planning/behavior_path_goal_planner_module/src/goal_planner_module.cpp

```diff
@@ -277,49 +316,76 @@ void GoalPlannerModule::onTimer()
   } else {
     RCLCPP_ERROR(
       getLogger(), "path_priority should be efficient_path or close_goal, but %s is given.",
-      parameters_->path_priority.c_str());
+      parameters.path_priority.c_str());
     throw std::domain_error("[pull_over] invalid path_priority");
   }
 
   // set member variables
-  {
-    const std::lock_guard<std::recursive_mutex> lock(mutex_);
-    thread_safe_data_.set_pull_over_path_candidates(path_candidates);
-    thread_safe_data_.set_closest_start_pose(closest_start_pose);
-    RCLCPP_INFO(getLogger(), "generated %lu pull over path candidates", path_candidates.size());
-  }
+  thread_safe_data_.set_pull_over_path_candidates(path_candidates);
+  thread_safe_data_.set_closest_start_pose(closest_start_pose);
+  RCLCPP_INFO(getLogger(), "generated %lu pull over path candidates", path_candidates.size());
 
-  last_previous_module_output_ = previous_module_output;
+  thread_safe_data_.set_last_previous_module_output(previous_module_output);
 }
 
 void GoalPlannerModule::onFreespaceParkingTimer()
 {
   const ScopedFlag flag(is_freespace_parking_cb_running_);
 
-  if (getCurrentStatus() == ModuleStatus::IDLE) {
+  std::shared_ptr<const PlannerData> local_planner_data{nullptr};
+  std::optional<ModuleStatus> current_status_opt{std::nullopt};
+  std::shared_ptr<OccupancyGridBasedCollisionDetector> occupancy_grid_map{nullptr};
+  std::optional<GoalPlannerParameters> parameters_opt{std::nullopt};
+  std::shared_ptr<GoalSearcherBase> goal_searcher{nullptr};
+
+  // begin of critical section
+  {
+    std::lock_guard<std::mutex> guard(gp_planner_data_mutex_);
+    if (gp_planner_data_) {
+      const auto & gp_planner_data = gp_planner_data_.value();
+      local_planner_data = std::make_shared<const PlannerData>(gp_planner_data.planner_data);
+      current_status_opt = gp_planner_data.current_status;
+      occupancy_grid_map = gp_planner_data.occupancy_grid_map;
+      parameters_opt = gp_planner_data.parameters;
+      goal_searcher = gp_planner_data.goal_searcher;
```

Comment:
currently this goal_sercher is not used.

Reply:
I needed to use this in planFreespacePath()

Reply:
f100ba08df877b81a8f06dd1aa20332ab109cac8